### DISCUSSION TITLE
Next build doesn't complete

### DISCUSSION THREAD

**Discussion by amirhega** [vercel/next.js#53663] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:22 PDT 2023; root:xnu-8796.121.3~7/RELEASE_X86_64
    Binaries:
      Node: 20.5.0
      npm: 8.19.4
      Yarn: N/A
      pnpm: N/A
    Relevant Packages:
      next: 13.4.13-canary.16
      eslint-config-next: 13.4.12
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 4.9.5
    Next.js Config:
      output: N/A.
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

ESLint (eslint-config-next), Package manager (npm, pnpm, Yarn), TypeScript (plugin, built-in types)

### Link to the code that reproduces this issue or a replay of the bug

cant

### To Reproduce

No steps to reproduce it just started happening

### Describe the Bug

Running next build, next format and such stall and never finish. left over night and still never finished. Format kills the terminal. Works on Vercel

### Expected Behavior

Build to execute and format to execute. 

### How are you deploying your application? (if relevant)

Vercel

**Comment by icyJoseph** [comment]
Have you got any other dependencies installed? There's often packages that use deeply nested, or recursive types, and those can sometimes run into very strange issues where they never complete.
